src/main/java/org/spongepowered/api/entity/ai/task/AbstractAITask.java

```diff
@@ -36,16 +36,25 @@
  *
  * It is required for anyone wanting to write their own logic that a Goal can
  * run to utilize this class. If you desire to use the builtin AI included with
- * Minecraft, use {@link GameRegistry#createBuilder(Class)} and pass a builder to
- * it instead.
+ * Minecraft, use {@link GameRegistry#createBuilder(Class)} and pass a builder
+ * to it instead.
  *
  * @param <O> The type of Agent
  */
 public abstract class AbstractAITask<O extends Agent> implements AITask<O> {
 
     private final AITaskType type;
 
-    public AbstractAITask(AITaskType type) {
```

Comment:
More exactly, this is like the declaration of `TypeToken`, having a constructor `protected TypeToken() {}` for subclasses.